src/backend/distributed/operations/repair_shards.c

```diff
@@ -1063,12 +1063,29 @@ EnsureShardCanBeCopied(int64 shardId, const char *sourceNodeName, int32 sourceNo
 	{
 		if (targetPlacement->shardState == SHARD_STATE_TO_DELETE)
 		{
-			ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-							errmsg(
-								"shard " INT64_FORMAT " already exists in the target node",
-								shardId),
-							errdetail(
-								"The existing shard is marked for deletion, but could not be deleted because there are still active queries on it")));
+			/*
+			 * Trigger deletion of orphaned shards and hope that this removes
+			 * the shard.
+			 */
+			DropMarkedShardsInDifferentTransaction();
+			shardPlacementList = ShardPlacementList(shardId);
```

Comment:
are we guaranteed that the change is in the snapshot here?

Reply:
We indeed do call `AcceptInvalidationMessages`. Also the tests pass.